web/src/settings_org.js

```diff
@@ -262,21 +264,51 @@ function set_msg_move_limit_setting(property_name) {
     enable_or_disable_related_message_move_time_limit_setting(property_name, disable_setting);
 }
 
-function message_delete_limit_setting_enabled(setting_value) {
+function message_delete_limit_setting_enabled() {
     // This function is used to check whether the time-limit setting
-    // should be enabled. The setting is disabled when delete_own_message_policy
-    // is set to 'admins only' as admins can delete messages irrespective of
-    // time limit.
-    if (setting_value === settings_config.common_message_policy_values.by_admins_only.code) {
+    // should be enabled. The setting is disabled when every user
+    // who is allowed to delete their own messages is also allowed
+    // to delete any message in the organization.
+    const realm_delete_own_message_policy = Number.parseInt(
+        $("#id_realm_delete_own_message_policy").val(),
+        10,
+    );
+    let realm_can_delete_any_message_group_id = realm.realm_can_delete_any_message_group;
+    if (selected_realm_can_delete_any_message_group) {
+        realm_can_delete_any_message_group_id = selected_realm_can_delete_any_message_group;
+    }
+    const realm_can_delete_any_message_group_name = user_groups.get_user_group_from_id(
+        realm_can_delete_any_message_group_id,
+    ).name;
+    const common_message_policy_values = settings_config.common_message_policy_values;
+
+    if (realm_delete_own_message_policy === common_message_policy_values.by_admins_only.code) {
+        return false;
+    }
+    if (realm_can_delete_any_message_group_name === "role:administrators") {
+        return true;
+    }
+    if (realm_delete_own_message_policy === common_message_policy_values.by_moderators_only.code) {
+        return false;
+    }
+    if (realm_can_delete_any_message_group_name === "role:moderators") {
+        return true;
+    }
+    if (realm_delete_own_message_policy === common_message_policy_values.by_full_members.code) {
+        return false;
+    }
+    if (realm_can_delete_any_message_group_name === "role:fullmembers") {
+        return true;
+    }
+    if (realm_delete_own_message_policy === common_message_policy_values.by_members.code) {
```

Comment:
for this I guess you can use `user_groups.get_recursive_subgroups(realm_can_delete_any_message_group)` and check if `realm_delete_own_message_policy` in the returned set?

Reply:
We cannot check `realm_delete_own_message_policy` as subgroup for `realm_can_delete_any_message_group` because `realm_delete_own_message_policy` is not based on new group based permissions. It uses `CommonMessagePolicyEnum` rather than `UserGroup` for defining role based permission.

Reply:
ok, makes sense.